lib/src/package.dart

```diff
@@ -325,7 +314,16 @@ class Package {
               );
       },
       isDir: (dir) => dirExists(resolve(dir)),
-    ).map(resolve).toList();
+    ).map(resolve).map((path) {
```

Comment:
I think resolution of symlinks should happen inside `listDir: () {`, otherwise we will not get the pubignore/gitignore behavior of nested symlinks.

Reply:
this is resolution from platform-independent path (used in ignore files) to OS-dependent files

Reply:
Sorry I mean the last mapping (what you added)

The structure you do now is (very simplified):
```dart
Ignore.listFiles(
  listDir: (dir) => Directory(dir).listSync(followLinks: False),
  [...]
).map((path) => Link(path).targetSync)
```
And I suggest the logic should look more like:
```dart
Ignore.listFiles(
  listDir: (dir) => Directory(dir).listSync(followLinks: False)
      .map((path) => Link(path).targetSync),
  [...]
)
```

Reply:
Oh, I understand now

`followLinks: False` doesn't  get resolved as `Directory`

I changed this code to be consistent on windows and posix-like systems

I faced the following error with `followLinks: true` on windows
https://github.com/dart-lang/pub/runs/5048998732?check_suite_focus=true#step:7:58
<details>
  <summary>error</summary>

```
Expected: throws <Instance of 'DataException'> with `message`: contains 'Pub does not support publishing packages with non-resolving symlink:'
    Actual: <Closure: () => List<String>>
     Which: threw FileSystemException:<FileSystemException: Directory listing failed, path = 'C:\Users\runneradmin\AppData\Local\Temp\dart_test_5e4ddcd9\myapp\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\subdir\symlink\*' (OS Error: The system cannot find the path specified.
                  , errno = 3)>
            stack dart:io                                   _Directory.listSync
                  package:pub/src/package.dart 248:48       Package.listFiles.<fn>
                  package:pub/src/ignore.dart 291:28        Ignore.listFiles
                  package:pub/src/package.dart 245:19       Package.listFiles
                  test\package_list_files_test.dart 115:30  main.<fn>.<fn>
                  package:test_api                          expect
                  test\package_list_files_test.dart 114:5   main.<fn>
                  
            which is not an instance of 'DataException'
  
  package:test_api                         expect
  test\package_list_files_test.dart 114:5  main.<fn>

```

</details>


we don't really care if this is resolved as `Link` or `Directory`, we only interested in `path` field (and it would be the same either way)

i've just added additional check, hope it will work on windows as expected

Reply:
And we don’t care of exact target until we forced to print it in error message